sgt-back-end/index.js

```diff
@@ -0,0 +1,161 @@
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select "gradeId",
+         "name",
+         "course",
+         "score",
+         "createdAt"
+    from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An expected error occurred' });
+    });
+});
+
+app.post('/api/grades/', (req, res) => {
+  const postGrade = req.body;
+  const score = Number(postGrade.score);
+  if (!postGrade.name) {
+    res.status(400).json({ error: 'Must include a valid name' });
+  } else if (!postGrade.course) {
+    res.status(400).json({ error: 'Must include a valid course' });
+  } else if (!score) {
+    res.status(400).json({ error: 'Must include a valid score' });
+  } else if (score < 0 || score > 100 || !Number.isInteger(score)) {
+    res.status(400).json({ error: 'Score must be a positve integer between 0 & 100' });
+  } else {
+    const insertSql = `
+    insert into "grades" ("name", "course", "score")
+    values ('${postGrade.name}', '${postGrade.course}', '${score}')
+    returning *
+    `;
+
+    db.query(insertSql)
+      .then(result => {
+        const grade = result.rows[0];
+        res.status(201);
+        res.json(grade);
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({ error: 'An unexpected error has occurred.' });
+      });
+  }
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const gradeId = Number(req.params.gradeId);
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = req.body.score;
+
+  if (gradeId <= 0 || isNaN(gradeId)) {
+    res.status(400).send({ error: 'gradeId must be a postitve integer' });
```

Comment:
`gradeId`s should also trigger a `400 Bad Request` if they are a non-integer number.

Reply:
![gradeId-not-integer](https://user-images.githubusercontent.com/25161542/163086931-eb7b2dfc-d60a-455f-ae54-60df3d903acf.png)